packages/graph/hash_graph/lib/graph/src/store/mod.rs

```diff
@@ -30,4 +33,19 @@ pub use self::{
 ///
 /// In addition to the errors described in the methods of this trait, further errors might also be
 /// raised depending on the implementation, e.g. connection issues.
-pub trait Store = AccountStore + DataTypeStore + PropertyTypeStore + EntityTypeStore + EntityStore;
+#[async_trait]
+pub trait Store:
+    AccountStore + DataTypeStore + PropertyTypeStore + EntityTypeStore + EntityStore
+{
+    type Transaction<'t>: Transaction + Send
+    where
+        Self: 't;
```

Comment:
The syntax is ugly but sadly it's required currently. Even though GATs are stable, this may be lifted in the future.

Reply:
Worth a comment, or not really?